docs/administration-guide/gaffer-stores/federated-store.md

```diff
@@ -2,4 +2,406 @@
+
+When using the REST API, configuration is done with the files `graphConfig.json` and `store.properties`. Minimal examples are shown below.
+
+```json title="graphConfig.json"
+{
+  "graphId": "federatedGraph",
+  "library": {
+    "class": "uk.gov.gchq.gaffer.store.library.HashMapGraphLibrary"
+  }
+}
+```
+
+```properties title="store.properties"
+gaffer.store.class=uk.gov.gchq.gaffer.federatedstore.FederatedStore
+gaffer.store.properties.class=uk.gov.gchq.gaffer.federatedstore.FederatedStoreProperties
+```
+
+### Using the Java API
+
+A `Graph` Object containing this Federated Store graph can be created as follows:
+
+```java
+final Graph federatedGraph = new Graph.Builder()
+        .config(new GraphConfig.Builder()
+                .graphId("federatedGraph")
+                .library(graphLibrary) // (1)!
+                .build())
+        .storeProperties(new FederatedStoreProperties())
+        .build();
+```
+
+1.  [See GraphLibrary Javadoc for more info](https://gchq.github.io/Gaffer/uk/gov/gchq/gaffer/store/library/GraphLibrary.html).
+
+
+### Federated Store Properties
+
+In addition to the [standard Store Properties](store-guide.md#store-properties), there are some additional properties specific to this store:
+
+- `gaffer.federatedstore.isPublicAllowed`: Controls if adding graphs with [public access](#publicprivate-access) is allowed. True by default.
+- `gaffer.federatedstore.customPropertiesAuths`: String containing auths for [allowing users to use store properties other than those contained in a graph library](#limit-custom-properties). Unset by default, allowing all users to do this.
+- `gaffer.federatedstore.storeConfiguredMergeFunctions`: Path to file containing [merge function definitions to use to override the default merge functions](#setting-default-merge-functions).
+- `gaffer.federatedstore.storeConfiguredGraphIds`: Path to file containing [Graph IDs to be queried when a user doesn't target specific graphs](#setting-default-query-graph-ids). By default, all graphs a user can access are queried.
+
+??? example "Setting these properties with the Java API"
+
+    ```java
+    final FederatedStoreProperties exampleFederatedStoreProperties = new FederatedStoreProperties();
+    exampleFederatedStoreProperties.setGraphsCanHavePublicAccess(false);
+    exampleFederatedStoreProperties.setCustomPropertyAuths("Auth1, Auth2, Auth3");
+    exampleFederatedStoreProperties.setStoreConfiguredMergeFunctions("path/to/mergefunctionDefinitions.json");
+    exampleFederatedStoreProperties.setStoreConfiguredGraphIds("path/to/defaultGraphIds.json");
+    ```
+
+#### Setting default Merge Functions
+
+[Merge Functions](../../reference/operations-guide/federated-operation.md#the-merge-function) control how results from different sub-graphs in a Federated Store are merged together.
+If you want to override the default behaviour and use different merge functions, then you can do this using `storeConfiguredMergeFunctions`.
```

Comment:
Which file do you add this property to? Or is there a typo as you have `configuredMergeFunctions` in the example below not `storeConfiguredMergeFunctions`?

Reply:
This is pretty confusing. In the JSON file supplied via a store property `configuredMergeFunctions` is correct. But everywhere else it's called `storeConfiguredMergeFunctions`. This may be a typo in the Gaffer code which can be corrected in future to make this uniform.

Reply:
There is a PR to fix this: https://github.com/gchq/Gaffer/pull/3036

Reply:
This has now been merged into Gaffer and I have corrected these docs